packages/vetrina/src/Vetrina/Vetrina.purs

```diff
@@ -41,12 +41,14 @@ type JSProps =
   { onClose :: Nullable (Effect Unit)
   , onLogin :: Nullable (Effect Unit)
   , products :: Nullable (Array JSProduct)
+  , unexpectedError :: Nullable JSX
   }
 
 type Props =
-  { onClose  :: Effect Unit
-  , onLogin  :: Effect Unit
-  , products :: Either Error (Array Product)
+  { onClose         :: Effect Unit
+  , onLogin         :: Effect Unit
+  , products        :: Either Error (Array Product)
+  , unexpectedError :: Maybe JSX
```

Comment:
I think we should always expect to have this as prop (not `Maybe`), because there might be unexpected errors happening even when the parent isn't aware of them.

Reply:
And from the parent point of view, it should not even try to render vetrina if it knows something is wrong.

Reply:
I used the `Maybe` to discriminate in the `didMount` hook. What different strategy would you suggest?